### PR TITLE
control-plane-operator: add a CA for customer admin clients

### PR DESCRIPTION
**Pull request by stevekuznetsov**

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -164,6 +170,19 @@ type HostedControlPlaneReconciler struct {
 	awsSession                    *session.Session
 	reconcileInfrastructureStatus func(ctx context.Context, hcp *hyperv1.HostedControlPlane) (InfrastructureStatus, error)
 	NameServerIP                  string
+
+	KubeClient        kubernetes.Interface
+	OperatorClient    v1helpers.StaticPodOperatorClient
```
Comment (stevekuznetsov) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:175]: We will remove this when https://github.com/openshift/library-go/pull/1593 merges.

File: control-plane-operator/main.go [hunk 2]
```diff
@@ -388,7 +395,43 @@ func NewStartCommand() *cobra.Command {
...
 		}
 
-		if err := (&hostedcontrolplane.HostedControlPlaneReconciler{
+		kubeClient, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			setupLog.Error(err, "could not create k8s client")
+			os.Exit(1)
+		}
+		kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, namespace)
+		operatorClient, _, err := genericoperatorclient.NewStaticPodOperatorClient(restConfig, operatorv1.GroupVersion.WithResource("kubeapiservers")) // TODO: where do we report status?
```
Comment (stevekuznetsov) [inline control-plane-operator/main.go:404]: We will remove this when https://github.com/openshift/library-go/pull/1593 merges.

File: control-plane-operator/main.go [hunk 3]
```diff
@@ -388,7 +395,43 @@ func NewStartCommand() *cobra.Command {
...
+		controllerRef, err := librarygoevents.GetControllerReferenceForCurrentPod(ctx, kubeClient, namespace, nil)
+		if err != nil {
+			setupLog.Error(err, "unable to get owner reference (falling back to namespace)")
+			controllerRef = &corev1.ObjectReference{
+				Kind: "Namespace",
+				Name: namespace,
+			}
+		}
+		eventRecorder := librarygoevents.NewKubeRecorder(kubeClient.CoreV1().Events(namespace), "hostedcontrolplanecontroller", controllerRef)
+		certRotationScale, err := certrotation.GetCertRotationScale(ctx, kubeClient, namespace) // TODO: where to store this config?
```
Comment (stevekuznetsov) [inline control-plane-operator/main.go:418]: Does this even need to be configurable?

**Comment by enxebre** [comment]
Thanks! I'm trying to put together all the context here and wrap my head around it.  Would we get the same outcome that this PR is producing, if we added a new signer and cert for customer admin in kas_pki_setup? pretty much like our existing adminKubeconfigSigner but letting our default 30days validity be different.

```
	// signer
	adminKubeconfigSigner, err := reconcileSigner(
		manifests.SystemAdminSigner(hcp.Namespace),
		pki.ReconcileAdminKubeconfigSigner,
	)

	if err != nil {
		return err
	}
	totalClientCABundle = append(totalClientCABundle, adminKubeconfigSigner)

	// system:admin client cert
	if _, err := reconcileSub(
		manifests.SystemAdminClientCertSecret(hcp.Namespace),
		adminKubeconfigSigner,
		pki.ReconcileSystemAdminClientCertSecret,
	); err != nil {
		return err
	}
```

**Comment by stevekuznetsov** [comment]
@enxebre yep! I think it would be equivalent. I brought in `library-go` and the associated logic as @deads2k and @sjenning came to an agreement that it would be a useful approach, as it would (after some cleanup) have the side-effect of increasing the number of folks who could review & maintain the HyperShift PKI logic to everyone that understood the `library-go` bits, which are shared across the other parts of OCP.

**Pull request closed without merging**